File: trunk/tools/R_package_converter/hardcoded_wrappers/hardcoded_wrapper.R
AT.particle.name.from.particle.no <- function(particle.no){

     n                   <- length(particle.no)
     particle.name       <- character(n)
     
     for (i in 1:n){
          cur.particle.name     <- character(1)
          res                   <- .C( "AT_particle_name_from_particle_no_R",
		                               particle.no    = as.integer(particle.no[i]),           
									   particle.name  = as.character(cur.particle.name),
									   PACKAGE        = "libamtrack")
          particle.name[i]     <-     res$particle.name
     }          
     return(particle.name)
}
     

AT.particle.no.from.particle.name <- function(particle.name){

     n                    <- length(particle.name)
     particle.no          <- numeric(n)
     
     for (i in 1:n){
          cur.particle.no     <- numeric(1)
          res                 <- .C( "AT_particle_no_from_particle_name_R",
		                             particle.name  = as.character(particle.name[i]),  
                                     particle.no    = as.integer(cur.particle.no),
									 PACKAGE        = "libamtrack")
          particle.no[i]     <-     res$particle.no
     }          
     return(particle.no)
}

AT.material.name.from.material.no <- function(material.no){

     n                   <- length(material.no)
     material.name       <- character(n)
     
     for (i in 1:n){
          cur.material.name     <- character(1)
          res                   <- .C( "AT_material_name_from_material_no_R",
		                               material.no    = as.integer(material.no[i]),           
									   material.name  = as.character(cur.material.name),
									   PACKAGE        = "libamtrack")
          material.name[i]     <-     res$material.name
     }          
     return(material.name)
}
     

AT.material.no.from.material.name <- function(material.name){

     n                    <- length(material.name)
     material.no          <- numeric(n)
     
     for (i in 1:n){
          cur.material.no     <- numeric(1)
          res                 <- .C( "AT_material_no_from_material_name_R",
		                             material.name  = as.character(material.name[i]),  
                                     material.no    = as.integer(cur.material.no),
									 PACKAGE        = "libamtrack")
          material.no[i]     <-     res$material.no
     }          
     return(material.no)
}

File: trunk/tools/R_package_converter/hardcoded_wrappers/hardcoded_wrapper.c
#include "hardcoded_wrapper.h"

void AT_particle_name_from_particle_no_R(const int* particle_no,
    char** particle_name){

  const long n = 1;
  const long particle_no_long = (long)(*particle_no);
  char particle_name_str[1][6];

  strcpy(particle_name_str[0], *particle_name);

  AT_particle_name_from_particle_no( n,
      &particle_no_long,
      particle_name_str);

  strcpy(*particle_name, particle_name_str[0]);
}


void AT_particle_no_from_particle_name_R(const char** particle_name,
    int* particle_no){
  const long n = 1;
  char * particle_name_str = (char*)calloc(PARTICLE_NAME_NCHAR, sizeof(char));
  strcpy(particle_name_str, *particle_name);

  long particle_no_long = 0;

  AT_particle_no_from_particle_name( n,
      &particle_name_str,
      &particle_no_long);

  free( particle_name_str);

  *particle_no = (int)particle_no_long;
}

void AT_material_name_from_material_no_R( const int* material_no,
    char** material_name){

	const long material_no_long = (long)(*material_no);
	char * material_name_str = (char*)calloc(MATERIAL_NAME_LENGTH, sizeof(char));

	AT_material_name_from_number( material_no_long,
	      material_name_str);

	strcpy(*material_name, material_name_str);

	free(material_name_str);
}

void AT_material_no_from_material_name_R( const char** material_name,
	int* material_no){

	  char * material_name_str = (char*)calloc(MATERIAL_NAME_LENGTH, sizeof(char));
	  strcpy(material_name_str, *material_name);

	  long material_no_long = 0;

	  material_no_long	= AT_material_number_from_name( material_name_str);

	  free( material_name_str);

	  *material_no = (int)material_no_long;
}

